packages/@aws-cdk/aws-appsync/lib/schema.ts

```diff
@@ -34,13 +43,23 @@ export class Schema {
   /**
    * Generate a Schema from file
    *
-   * @returns `SchemaAsset` with immutable schema defintion
+   * @returns `Schema` with immutable schema defintion
    * @param filePath the file path of the schema file
    */
   public static fromAsset(filePath: string): Schema {
     return new Schema({ filePath });
   }
 
+  /**
+   * Point GraphQL Schema to a S3 Location
+   *
+   * @returns `Schema` with immutable schema defintion
+   * @param s3Location the s3 location of the schema
+   */
+  public static fromS3Location(s3Location: string): Schema {
```

Comment:
I wonder if just a string is a good API here. We often represent S3 locations as pairs of `IBucket` and object key.

We also have the [`Location` interface in the S3 module](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-s3.Location.html). Maybe that's a better abstraction than just a lonely string?

Reply:
hm totally right.. ill use the IBucket and object key method alongside `s3UrlForObject`